src/impl/transact_log_handler.cpp

```diff
@@ -497,6 +499,13 @@ class TransactLogObserver : public TransactLogValidationMixin, public MarkDirtyM
         return true;
     }
 
+    bool add_row_with_key(size_t row_ndx, size_t prior_num_rows, size_t key_col, int64_t)
+    {
+        insert_empty_rows(row_ndx, 1, prior_num_rows, false);
+        m_active_table->modify(row_ndx, key_col);
```

Comment:
This shouldn't be marking the row as modified (and there should be tests for the new instruction).